doc/users.md

```diff
@@ -1 +1,872 @@
 # Users Module
+Module | HTTP Method | URL | Description 
+--- | --- | --- | ---
+[Add User](#add) | POST | /add | Add User
+[Edit User](#edit) | PUT | /edit | Edit User
+[View User](#view) | GET | /view | View User
+[Delete User](#delete) | DELETE | /delete | Delete User
+
+## <a name="add"></a>Add Data User
+
+### Endpoint 
+POST /add
+
+### Database
+![](./user-layer.png)
+
+for add, you need get :
+- company_id, company_code, company_name from companies tables
+- branch_id, branch_code, branch_name from branches tables
+- region_id, region_code from regions tables
+- kios_id, kios_code from kios tables
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+name | string | dennis  
+username | string | vdennis  
+password | string | cdcgi2020  
```

Comment:
POST /users will be call by admin right?
it's need no password in the request payload. prefer generate random password and sent to users's email.